Lib/multiprocessing/pool.py

```diff
@@ -437,13 +464,26 @@ def _handle_workers(cache, taskqueue, ctx, Process, processes, pool,
                         inqueue, outqueue, initializer, initargs,
                         maxtasksperchild, wrap_exception):
         thread = threading.current_thread()
+        setattr(thread, '_state_lock', ctx.Lock())
```

Comment:
Why use `setattr` and not `thread._state_lock = ctx.Lock()`?

Reply:
I agree that would work, but since `_state_lock` is not part of the thread class definition I feel that `setattr` is more explicit about setting new members dynamically.

Since `_state_lock` is set during initialization of the worker handler, I then assume that _we know ahead of time_ that `_state_lock` is a member of the thread and access the lock directly using `thread._state_lock` instead of `getattr` (also for readability reasons).

Reply:
The `setattr` idiom you're talking about isn't used elsewhere in this module, so I'd rather stick to a normal attribute setting.

Reply:
After the rebasing this was not necessary anymore.